### ISSUE TITLE
is compose run supposed to work on fargate?

### ISSUE THREAD

**Issue by jpaas** [aws/amazon-ecs-cli#625]
### Summary
I can create a fargate cluster and deploy a compose file to create a service, but I can't use compose to run a single task using the same configuration.

### Description
I can deploy a service with 2 tasks, web and worker like so: 
```
ecs-cli compose --ecs-profile karma --cluster-config karma-backend-staging --file ./deploy/docker-compose.staging.yml --ecs-params ./deploy/ecs-params.staging.yml --project-name karma-backend-staging service up --create-log-groups
```

But if I try to run a one-time task using:
```
ecs-cli compose --ecs-profile karma --cluster-config karma-backend-staging --file ./deploy/docker-compose.staging.yml --ecs-params ./deploy/ecs-params.staging.yml run worker "echo test"
```

I get:
```
INFO[0001] Using ECS task definition                     TaskDefinition="deploy:5"
ERRO[0001] Error running tasks                           error="InvalidParameterException: Network Configuration must be provided when networkMode 'awsvpc' is specified.\n\tstatus code: 400, request id: 2828d597-c717-11e8-bd7e-71fd0be676be" task definition="arn:aws:ecs:us-east-1:023742962703:task-definition/deploy:5"
```
### Config files
docker-compose.yml:
```
version: "3"

services:
  web:
    command: bundle exec rails s
    env_file: .env.docker.staging
    environment:
      - SERVICE_NAME=web
    image: 023742962703.dkr.ecr.us-east-1.amazonaws.com/karma_backend_rails:test2
    logging:
      driver: awslogs
      options:
        awslogs-group: staging
        awslogs-region: us-east-1
        awslogs-stream-prefix: staging
    ports:
      - "80:80"

  worker:
    command: bundle exec sidekiq
    env_file: .env.docker.staging
    environment:
      - SERVICE_NAME=worker
    image: 023742962703.dkr.ecr.us-east-1.amazonaws.com/karma_backend_rails:test2
    logging:
      driver: awslogs
      options:
        awslogs-group: staging
        awslogs-region: us-east-1
        awslogs-stream-prefix: staging

```
ecs-params.yml:
```
version: 1
task_definition:
  ecs_network_mode: awsvpc
  task_execution_role: ecsTaskExecutionRole
  task_size:
    mem_limit: 0.5GB
    cpu_limit: 256
run_params:
  network_configuration:
    awsvpc_configuration:
      subnets:
        - "subnet-0fff0d5d419f9b111"
        - "subnet-02fc15df22223e4c2"
        - "subnet-07ce2e735e29aa4f8"
      security_groups:
        - "sg-01e748fdbbc7312db"
      assign_public_ip: DISABLED

```
~/.ecs/config:
```
version: v1
default: karma-backend-staging
clusters:
  karma-backend-staging:
    cluster: karma-backend-staging
    region: us-east-1
    default_launch_type: FARGATE
```

**Comment by allisaurus** [comment]
Hi @jpaas - you should definitely be able to run a singe task with FARGATE. The network configuration in your `ecs-params.yml` looks correct, so w/o knowing the specifics re: the subnets & security group you're sending in, it's not clear why this would fail. 

I haven't been able to reproduce the error you provided, so at the risk of asking you to take steps you've already done:

1. Can you confirm that the contents of the ecs-params.yml you provided matches exactly what you're sending in? (e.g., the `./deploy/ecs-params.staging.yml` file)?
2. Can you try running `ecs-cli compose up` (i.e., sans overrides) and let me know if that fails as well?
3. Confirm which version of the ecs-cli you're using? (1.8.0 is latest)

Thanks!

**Comment by jpaas** [comment]
Hi @allisaurus ...
1. Yes those are the exact files I am using
2. compose up worked
3. 1.7.0

**Comment by allisaurus** [comment]
Thanks for the additional info! I was able to reproduce the error with 1.7.0. 

Since that we release, we made [some changes](https://github.com/aws/amazon-ecs-cli/commit/d8fc41ce469fcd1a08eacdc222f8d44db0075f28) to RunTask which allow a network configuration to be passed through. If you upgrade to **1.8.0** (download links are in our README under "Latest"), this should be fixed. 

Please let me know if you upgrade and continue to have problems!

**Comment by jpaas** [comment]
Hi @allisaurus it does work with 1.8.0. I had installed it using brew and they haven't released 1.8.0 yet. Thanks!

Is there any way to run an interactive shell? I have asked this question on the AWS forum, but no answer yet.

**Comment by allisaurus** [comment]
@jpaas  Great, thanks for letting us know it's working for you. 

Re: interactive shell, there isn't currently a way to launch an interactive shell within a Fargate task. CloudWatch logs (which you have configured above) or the [task stopped reason](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/stopped-task-errors.html) are probably your best resources for debugging problems you might have launching/maintaining these tasks.
